cli/src/main/java/com/devonfw/tools/ide/cli/Ide.java

```diff
@@ -146,6 +265,22 @@ private CliArgument initContext(CliArgument first) {
       }
       current = current.getNext(true);
     }
+    return current;
+  }
+
+  private CliArgument retrieveCliArgumentByContext(CliArgument first, ContextCommandlet contextCommandlet) {
+
+    CliArgument current = retrieveCliArgument(first, contextCommandlet);
+    if (contextCommandlet.version.isTrue()) {
```

Comment:
FYI: Maybe it would have been easier to create a separate `VersionCommandlet` that can do nothing but print the version and exit.

Reply:
I've removed -v and --version and replaced it with VersionCommandlet now.